core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -293,6 +293,15 @@ public void initializeWithCustomizedUgi() throws Exception {
     org.apache.hadoop.fs.FileSystem.get(uri, conf);
   }
 
+  @Test
+  public void initializeWithFullPrincipalUgi() throws Exception {
+    mockUserGroupInformation("testuser@ALLUXIO.COM");
+
+    final org.apache.hadoop.conf.Configuration conf = getConf();
+    URI uri = URI.create(Constants.HEADER + "host:1");
+    org.apache.hadoop.fs.FileSystem.get(uri, conf);
+  }
```

Comment:
Is there anyway to avoid relying on that implementation throwing an exception, is there any state we can check explicitly to ensure the right thing has happened?